HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+- Information to the user: With this mechanism described in this document, the Airdrop-Sender will have to check their unsolicited-outbox to see which tokens have not been deposited to the Airdrop-Recipients’ accounts; and the airdrop-Recipient will have to check their unsolicited-inbox to see which tokens are pending their acceptance.
+- Wallets, Exchanges, and mirror-node explorers will have to create visibility and notification mechanisms for the users.
+        
+## Backward Compatibility
+TBD
+## Security Implications
+        
+- User security: Since no tokens show up in a user's account without their explicit approval, their security is not compromised in any way.
+- Network security: Since the cost of the operations and usage of network resources such as storage of the tokens in the unsolicited-inbox is paid for by the sender of the airdrop tokens, the network security is not compromised either. These are not 'rejected' ideas *per se*. Instead, these ideas were considered and we felt that these do not qualify to be the highest priority feature and that these can be implemented in the future.
+    
+## How To Teach This
+TBD
+## Reference Implementation
+TBD
+## Rejected Ideas
+    
+- Wallet-based/Mirror-node-based state: Implement this functionality primarily on the mirror node as follows. Allow all token transfers to go through - even if the user has not associated the token. When the user (Airdrop-Recipient) sees these unwanted tokens in their accounts, then they register with the wallet or mirror node that they don't want to see the tokens in their account. And the mirror nodes never report those tokens after receiving that request. This idea doesn't work because if the user uses a different wallet or a different mirror node, then they will still see all objectionable tokens. Also, they will have to pay for the renewal of these token associations even if they don't want the tokens.
+- Maintain the spam folder only on the Airdrop-recipient's side. In other words, allow anybody to send any tokens to a recipient's account, but the tokens go into a spam folder. And if they don't accept it for a given period of time, then delete or return the tokens to the sender. This approach was not accepted because this will put the burden of deleting the token as well as the temporary storage/renewal fees on the innocent recipient without providing any disincentives to the Airdrop-Sender. The proposed scheme makes the sender pay for these charges.
+    
+## Open Issues/FAQ
+Q1: Expected behavior when the user is not associated with tokens but the wallet has auto-associate slots available ? Does the receiver pay for association ? If the auto-associate slot is used for air-drop. does the spammer still pay for a return token ? Can the airdrop tokens received against associated slot be returned ? Can all token be returned ?
+
+Q2: Would we then suggest that receiver never have auto-associate enabled.
```

Comment:
I think there are use cases that may still want any token sent to the wallet. Even if accounts or treasury accounts can be whitelisted, some wallets still may want to auto accept tokens no matter where they come from.

auto-association call may just have to be tweaked in marketing /usage suggestions with the inbox. Positioned as auto accept x amount of tokens that are sent to the inbox. The payment for return token gas from the sender would return to sender automatically when a auto-association slot was used

Reply:
Hmm, I think the concept of auto-association would sit besides and not inside that our the inbox in my mind. 
By setting auto association you are saying you are willing to associate with anything sent to you, in which case the network need not consider token inbox logic.
The inbox would come into play when no auto-association slots exist